business_requirement_deliverable/models/business.py

```diff
@@ -11,7 +11,7 @@ class BusinessRequirementDeliverable(models.Model):
     _name = "business.requirement.deliverable"
     _inherit = ["mail.thread", "mail.activity.mixin", "portal.mixin"]
     _description = "Business Requirement Deliverable"
-    _order = "business_requirement_id, section_id, sequence, id"
+    _order = "business_requirement_id, section_id desc, sequence, id"
```

Comment:
Why changing this?

Reply:
Check my other comment that solves this.

Reply:
As far I understand it is needed to create first the deliverable lines with undefined section.
Please check this screenshot:
![screenshot-127 0 0 1_12069-2019 08 22-19_48_03](https://user-images.githubusercontent.com/16461806/63537420-feb87f80-c515-11e9-81aa-7532fc91d245.png)
But is inside the section Optional

Was the main reason I did with filtered method, but with the order you show was possible done on the same loop

Reply:
OK, I see, I haven't though on this. We have 3 possible solutions:

- Create a section line "[No section]" (translated).
- Call `br.deliverable_ids.sorted(...` with that order for having first the ones with no section. But I don't like it because the rest of the sections are not respected. I'm also thinking that we have a sequence in sections, isn't it? (or if not, we should), for sorting how do we want to put sections. This is not respected when we put `section_id` in the order, but there's no way to do it in the `_order` attribute. We must do it through the sorted function. We can specify the sorting method for putting first the no section and then sort by section sequence.
- Split in 2 the loop, but we duplicate code. One variant is to pre-sort manually on a populated recordset our deliverables and then do the loop, but previous sorted custom method seems cleaner.

Reply:
sorted method works fine by default is getting the undefined sections then first section in order though we are done here except for the migration script